src/shogun/io/SGIO.cpp

```diff
@@ -189,6 +196,7 @@ void SGIO::progress(
 			if (current_val >= max_val-1)
 			{
 				v = 100;
+				last_progress=v-1e-6;
```

Comment:
Are you referring to `last_progress` or `v`? If you're referring to `last progress` I think of -1 as a value that means _not initialized_ and I used it since there are cases in which `last_progress` is not set (for example when `max_val-min-val<=0`). Obviously, this is completely subjective. Using 0 instead of -1 should not make any difference. :)